tools/build-id.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python3
+
+import os
+import subprocess
+import urllib.request
+import zipfile
+import argparse
+
+READELF_TOOL_PATH="/toolchains/llvm/prebuilt/darwin-x86_64/bin/x86_64-linux-android-readelf"
+
+FLAVORS = ['base', 'objectServer']
+ARCHS = ['arm64-v8a', 'armeabi-v7a', 'x86', 'x86_64']
+
+def file_exists(path):
+    f = None
+    try:
+        f = open(path)
+        return True
+    except IOError:
+        return False
+    finally:
+        if f:
+            f.close()
+
+def find_ndk():
+    PROPS_FILE=f"{os.getcwd()}/../realm/local.properties"
+    if not file_exists(PROPS_FILE):
+        print(f"{PROPS_FILE} not found! NDK location cannot be determined")
+        exit()
+    
+    return subprocess.check_output(f'grep "ndk.dir" "{PROPS_FILE}" | cut -d = -f2', shell=True).decode('ascii').strip()
+
+
+BUILD_ID_EXTRACT_COMMMAND = find_ndk() + READELF_TOOL_PATH + ' -n {0} | grep "Build ID" | cut -d ":" -f2'
+
+def download_file(url, path):
+    urllib.request.urlretrieve(url, path)
+
```

Comment:
Whats with all the empty lines. Is it PEP08 dictated?

Reply:
It is two empty lines spaces between functions..